### PR TITLE
Infix to postfix

### LINKED ISSUE #969: infix expression to postfix expression

### PR DESCRIPTION
**Pull request by raghav-dalmia**
Fixes #969 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Sample Input / Output.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

#### Changes proposed in this pull request:

- add the code of infix to postfix conversion

#### Languages Used:
- c++

#### Files Added:
- Infix_To_Postfix.cpp

@jainaman224, please review. Thank you

### CODE REVIEW

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		temp = optr.top();
+	    }
+			optr.pop(); // pop '(' from stack
+	}
+	else
+	{
+	    int precedence_of_curr_char = precedence_of_operator(input[i]); // get precedence of next char of string
+	    if(precedence_of_curr_char!=0) // if next char is an operator
+	    {
+		if(optr.size()==0) // if size of stack is zero
```
Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: add space around !=

Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: also around ==

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		    char prev_operator = optr.top(); // previous char in stack
+		    if(precedence_of_operator(prev_operator)<precedence_of_curr_char)
+		    optr.push(input[i]);
+		    else
+		    {
+			while(optr.size()!=0)
+		        {
+			    if(optr.top()=='(')
+				break;
+			    output+=optr.top(); // append all operator in output string that presents in the stack or in parenthesis
```
Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: space around == and != operators

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			optr.push(input[i]);
+		    }
+		}
+	    }
+	    else // if next character of input string is an operand
+	    {
+		output+=input[i]; // append that operand in the output string
+	    }
+	}
+	i++; // increase the index of string
```
Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: space around +=

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    // if we had traversed through complete string
+    // then append all operator in output string that presents in the stack 
+    while(optr.size()!=0)
+    {
+	output+=optr.top();
+	optr.pop();
+    }
```
Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: around != and around +=

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    return output;
+}
+
+int main()
+{
+    string input;
+    getline(cin,input);
+    string output = infix_to_postfix(input);
+    cout<<output<<endl;
+    return 0;
```
Comment (KanchanThareja) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: add space after cin, and before input
also around <<

File: Infix_To_Postfix/Infix_To_Postfix.cpp (hunk 2 above)
Comment (KanchanThareja) [reply Infix_To_Postfix/Infix_To_Postfix.cpp]: also after output and before optr.top

File: Infix_To_Postfix/Infix_To_Postfix.cpp [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    if(optr == '=')
+	return 1;
+    return 0; // return 0 if given character is not an operator
+}
+
+string infix_to_postfix(string input)
+{
+    string output = ""; // store postfix expression of given infix expression
+    stack<char> optr; // stack that stores all operators
+    int i=0;
```
Comment (somya-kapoor) [inline Infix_To_Postfix/Infix_To_Postfix.cpp]: Please add space around =

**Comment by raghav-dalmia** [comment]
@somya-kapoor @KanchanThareja done.

**Comment by raghav-dalmia** [comment]
@jainaman224, please merge this PR.

**Pull request merged**